x-pack/plugins/security_solution/public/timelines/components/timeline/body/renderers/formatted_field_udt.tsx

```diff
@@ -0,0 +1,452 @@
+    headers
+      ?.filter(
+        (h) => browserFieldsByName[h.id] && browserFieldsByName[h.id].format === BYTES_FORMAT
+      )
+      .map((c) => c.id) ?? [];
+  return [
+    ...PORT_NAMES,
+    EVENT_DURATION_FIELD_NAME,
+    HOST_NAME_FIELD_NAME,
+    USER_NAME_FIELD_NAME,
+    SIGNAL_RULE_NAME_FIELD_NAME,
+    EVENT_MODULE_FIELD_NAME,
+    SIGNAL_STATUS_FIELD_NAME,
+    AGENT_STATUS_FIELD_NAME,
+    RULE_REFERENCE_FIELD_NAME,
+    REFERENCE_URL_FIELD_NAME,
+    EVENT_URL_FIELD_NAME,
+    INDICATOR_REFERENCE,
+    ...ipFieldTypeFields,
+    ...geoFieldTypeFields,
+    ...byteFormattedFields,
+    ...columnNamesNotDraggable,
+  ].reduce(
+    (
+      obj: Record<string, (props: EuiDataGridCellValueElementProps) => React.ReactNode>,
+      field: string
+    ) => {
+      obj[field] = FieldValueCell(dataTableRows, scopeId, browserFieldsByName, closeCellPopover);
+      return obj;
+    },
+    {}
+  );
+};
+
+const FormattedFieldValueComponent: React.FC<{
+  asPlainText?: boolean;
+  /** `Component` is only used with `EuiDataGrid`; the grid keeps a reference to `Component` for show / hide functionality */
+  Component?: typeof EuiButtonEmpty | typeof EuiButtonIcon;
+  contextId: string;
+  eventId: string;
+  isAggregatable?: boolean;
+  isObjectArray?: boolean;
+  fieldFormat?: string;
+  fieldName: string;
+  fieldType?: string;
+  isButton?: boolean;
+  isDraggable?: boolean;
+  onClick?: () => void;
+  onClickAriaLabel?: string;
+  title?: string;
+  truncate?: boolean;
+  value: string | number | undefined | null;
+  linkValue?: string | null | undefined;
+}> = ({
+  asPlainText,
+  Component,
+  contextId,
+  eventId,
+  fieldFormat,
+  isAggregatable = false,
+  fieldName,
+  fieldType = '',
+  isButton,
+  isObjectArray = false,
+  isDraggable = true,
+  onClick,
+  onClickAriaLabel,
+  title,
+  truncate = true,
+  value,
+  linkValue,
+}) => {
+  if (isObjectArray || asPlainText) {
+    return <span data-test-subj={`formatted-field-${fieldName}`}>{value}</span>;
+  } else if (fieldType === IP_FIELD_TYPE) {
+    return (
+      <FormattedIp
+        Component={Component}
+        eventId={eventId}
+        contextId={contextId}
+        fieldName={fieldName}
+        fieldType={fieldType}
+        isAggregatable={isAggregatable}
+        isButton={isButton}
+        isDraggable={isDraggable}
+        value={!isNumber(value) ? value : String(value)}
+        onClick={onClick}
+        title={title}
+        truncate={truncate}
+      />
+    );
+  } else if (fieldType === GEO_FIELD_TYPE) {
+    return <>{value}</>;
+  } else if (fieldType === DATE_FIELD_TYPE) {
+    const classNames = truncate ? 'eui-textTruncate eui-alignMiddle' : undefined;
+    return isDraggable ? (
+      <FormattedDate className={classNames} fieldName={fieldName} value={value} />
+    ) : (
+      <FormattedDate className={classNames} fieldName={fieldName} value={value} />
+    );
+  } else if (PORT_NAMES.some((portName) => fieldName === portName)) {
```

Comment:
We should check all these custom rendering types render properly as when I load a port field, this is what I see. `destination.port` and `client.port`

<img width="227" alt="Screenshot 2024-03-08 at 12 52 47 PM" src="https://github.com/elastic/kibana/assets/17211684/74bb62f9-4d38-4721-906d-eb0f65b5d168">

Reply:
Thank for pointing this out. Let me try to reproduce it. Because I was seeing port fields blank as expected.

![image](https://github.com/elastic/kibana/assets/7485038/990ba335-b10c-47f5-b8f1-145212abacdd)

Reply:
So I have removed most of the logic from this file in favour old `DefaultCellRender` logic. The functions in this file were doing same things again. See a141c2b1c6ca7f9dce37bbed0e0001793c808482.

Also, do you mind, testing it again. May be try with different data views.